nvtabular/loader/backend.py

```diff
@@ -212,23 +211,18 @@ def __init__(
         )
 
     def __iter__(self):
-        indices = cp.arange(self.dataset.to_ddf().npartitions)
         if self.shuffle:
-            cp.random.shuffle(indices)
-        for dev in self.devices:
-            itr = TensorBatchDatasetItrFactory().create(
-                self.dataset,
-                self.library,
-                indices=indices.tolist(),
-                device=dev,
-                total_devs=self.devices,
-            )
-            t1 = threading.Thread(target=self.buff.load_chunks, args=(dev, itr))
-            t1.daemon = True
-            t1.start()
+            cp.random.shuffle(self.itrs[0].indices)
```

Comment:
I dont think this will work for the shuffling... we need to have the same index after randomized in all iterators. Here you only shuffling one... so all other iterators will still grab a chunk, from the ordered list, right?

Reply:
Yeah this is someone I was meaning to talk to you about. I think this implementation should work fine since, at the moment, they all get passed the same `indices` object and so the in-place shuffle done by `cp.random.shuffle` shuffles the underlying object they're all pointing to (hence why I only shuffle the first `itrs`'s `indices`). That said, if the intention is to have `AsyncIterator`s ever instantiated by `TensorBatchDatasetItr` objects that *weren't* created in the `__init__` of a `DataLoader` object (and thus could have disparate `indices` attributes), then this would be problematic. I guess it depends on how much of this we think should just be internal.